src/structs.rs

```diff
@@ -1,84 +1,54 @@
 use std::time::Duration;
 use std::convert::TryInto;
+use rustix::time::{Itimerspec, Timespec};
 
 use TimerState;
-use libc;
 
-// libc timespec is really awkward to work with (no traits etc)
-// so we have our own
-#[repr(C)]
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
-struct timespec {
-    tv_sec: libc::time_t,
-    tv_nsec: libc::c_long,
-}
-
-#[repr(C)]
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct itimerspec {
-    it_interval: timespec,
-    it_value: timespec,
-}
-
-impl itimerspec {
-    pub fn null() -> itimerspec {
-        itimerspec { it_interval: TS_NULL, it_value: TS_NULL }
-    }
-}
-
-const TS_NULL: timespec = timespec { tv_sec: 0, tv_nsec: 0 };
+const TS_NULL: Timespec = Timespec { tv_sec: 0, tv_nsec: 0 };
 
-impl From<Duration> for timespec {
-    fn from(d: Duration) -> timespec {
-        timespec {
-            tv_sec: d.as_secs() as libc::time_t,
-            // This can theoretically fail on 32-bit (`u32` to `c_long = i32`)
-            tv_nsec: d.subsec_nanos().try_into()
-                .expect("timespec overflow when converting from Duration"),
-        }
+fn to_timespec(d: Duration) -> Timespec {
+    Timespec {
+        tv_sec: d.as_secs().try_into().unwrap(),
+        tv_nsec: d.subsec_nanos() as _,
     }
 }
 
-impl From<timespec> for Duration {
-    fn from(ts: timespec) -> Duration {
-        Duration::new(ts.tv_sec as u64, ts.tv_nsec.try_into()
-            .expect("timespec overflow when converting to Duration"))
-    }
+fn from_timespec(ts: Timespec) -> Duration {
+    Duration::new(ts.tv_sec as u64, ts.tv_nsec as u32)
```

Comment:
same as above

Reply:
This is effectively assuming Linux will never return a `Timespec` with a negative `tv_secs`. Linux considers such values to [be invalid](https://github.com/torvalds/linux/blob/56c244382fdb793986097df8e29f9f9320bc2c60/include/linux/time64.h#L91). I've now added a comment about this. But if you'd prefer to have the code be defensive here, I can add the overflow check back in.